main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -2253,11 +2253,11 @@ object Typer {
         val ms = methods map visitJvmMethod
         TypedAst.Expression.NewObject(name, clazz, tpe, pur, eff, ms, loc)
 
-      case KindedAst.Expression.NewChannel(exp, tpe, loc) =>
+      case KindedAst.Expression.NewChannel(exp, loc) =>
         val e = visitExp(exp, subst0)
         val pur = Type.Impure
         val eff = e.eff
-        TypedAst.Expression.NewChannel(e, Type.mkChannel(tpe, loc), pur, eff, loc)
+        TypedAst.Expression.NewChannel(e, Type.mkChannel(e.tpe, loc), pur, eff, loc)
```

Comment:
This seems OK

Reply:
Right, this is not OK.

Reply:
Here is what we need to do:
- KindedAst.Expression.NewChannel(exp, loc) needs to have an extra argument `elmType` which is a Type.Var (constructed with Type.fresh... but in the previous phase (or maybe in an even earlier phase-- check some of the other uses). Anyway, it will have this `elmType`.
- Here in reassembly the type of the channel will be `mkChannel(subst0(elmType))`.